### ISSUE TITLE
Avoid relying on sortability of incompatible types

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5328]
|[<img alt="pitrou's avatar" src="https://avatars.githubusercontent.com/u/1721820?s=50" width="50" height="50">](https://github.com/pitrou)| @pitrou reported|
|-|-|
|Trac ID|trac#5328|
|Type|enhancement|
|Created|2011-10-22 02:30:05Z|

test_dirdbm relies on sorting arbitrary types, which fails under Python 3:

```
>>> 1 < None
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unorderable types: int() < NoneType()
```

It is reasonable to use a set instead. Patch attached.

Attachments:

* [test_dirdbm_sort.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/318/3188db3f9446e4820a6436507f089b1f3a836b6d/366b4ccca5fe5818b6faab7a228ae8af31cab2ca.patch) (1883 bytes) - added by antoine on 2011-10-22 02:30:14Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5328 5328
type__enhancement enhancement
reporter__antoine antoine
priority__normal normal
milestone__Python_3_x Python-3.x
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1319250605000000 1319250605000000
changetime__1320028042000000 1320028042000000
version__None None
owner__ 
cc__antoine
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_2"></a>|@itamarst set status to `closed`|
|-|-|

(In [33011]) Python 3 compatibility for test_dirdbm.py.

Fixes: [#5328](https://github.com/twisted/twisted/issues/5328)
Author: antoine
Review: itamar

Python 3 doesn't support sorting non-comparable types, so use set instead of sorted list.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

Presumably itamar meant to remove the review keyword, since it seems he reviewed this in order to decide to merge it.  I'll steal the credit to teach him a lesson about tracker hygiene.
